python/src/scipp/neutron/instrument_view.py

```diff
@@ -457,96 +611,259 @@ def update_colorbar(self):
+            colors[masks_inds] = masks_colors
 
-        self.geometry.attributes["color"].array = colors
+        self.geometry.attributes["color"].array = colors[:, :3]
 
         self.label.value = name_with_unit(
-            var=self.hist_data_array[self.key].coords[self.dim],
+            var=self.hist_data_array[self.key].coords[self.slider_dim],
             name=value_to_string(self.hist_data_array[self.key].coords[
-                self.dim].values[change["new"]]))
+                self.slider_dim].values[change["new"]]))
         return
 
     def change_projection(self, owner):
 
-        if owner.description == self.current_projection:
-            owner.button_style = "info"
-            return
-        if self.current_projection is not None:
+        projection = owner.description
+        axis = projection[-1]
+
+        # Just move the camera if we are staying as a 3D projection
+        if projection.startswith("3D") and self.current_projection.startswith(
+                "3D") and (not self.lock_camera):
+            new_cam_pos = [
+                self.camera_pos * (axis == "X"),
+                self.camera_pos * (axis == "Y"),
+                self.camera_pos * (axis == "Z")
+            ]
+            self.camera.position = new_cam_pos
             self.buttons[self.current_projection].button_style = ""
+            self.current_projection = projection
+            self.buttons[projection].button_style = "info"
+            self.update_colors({"new": self.slider.value})
+            return
 
-        projection = owner.description
+        if self.current_projection in self.buttons:
+            self.buttons[self.current_projection].button_style = ""
 
         # Compute cylindrical or spherical projections
         permutations = {"X": [0, 2, 1], "Y": [1, 0, 2], "Z": [2, 1, 0]}
-        axis = projection[-1]
 
-        if projection == "3D":
-            xyz = self.det_pos
+        if self.select_rendering.value == "Full":
+
+            # Duplicate the detector shape to the number of detectors by
+            # creating a Variable of dims ["spectrum", "vertex"]. The rotation
+            # operation will then be applied along the "spectrum" dimension
+            # using the automatic broadcast
+            vertices = sc.Variable(dims=[self.other_dim, "vertex"],
+                                   shape=[self.ndets, self.nverts],
+                                   unit=sc.units.m,
+                                   dtype=sc.dtype.vector_3_float64)
+            scaling = np.array(
+                self.hist_data_array[self.key].attrs["detector_shape"].values)
+            for i in range(self.nverts):
+                vertices["vertex", i] = sc.Variable(
+                    dims=[self.other_dim],
+                    values=np.tile(self.detector_shape[i], [self.ndets, 1]) *
+                    scaling,
+                    unit=sc.units.m,
+                    dtype=sc.dtype.vector_3_float64)
```

Comment:
Why is the tiling needed here? Can you make use of broadcasting?
```python
vertices['vertex', i] = self.detector_shape[i] * scaling
```

Or even reduce the whole thing to something like
```python
vertices = self.detector_shape * self.hist_data_array[self.key].attrs["detector_shape"]
```
?

Reply:
After investigation, this was not very easy to achieve so we will keep it as is for now.
The syntax might be easier in the future if we succeed in using Geometry Instancing (e.g. https://medium.com/@pailhead011/instancing-with-three-js-part-2-3be34ae83c57)